cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -212,14 +226,380 @@ data ProtocolParameters =
+  --     parameterExists maxValueSize (ProtocolParamsErrMustDefineMaxValueSize anyEra)
+
+  -- | Check if a protocol parameter was specified and fail if not.
+  parameterExists
+    :: Maybe a
+    -> ProtocolParametersError
+    -> Either ProtocolParametersError ()
+  parameterExists value err =
+    maybe (Left err) Right value >> Right ()
+
+
+newtype MaxTxExecutionUnits =
+    MaxTxExecutionUnits { unMaxTxExecutionUnits :: ExecutionUnits}
+    deriving (Eq, Show)
+
+_fromMaxTxExec :: Alonzo.ExUnits -> MaxTxExecutionUnits
+_fromMaxTxExec (Alonzo.ExUnits mMem mSteps) =
+  MaxTxExecutionUnits $ ExecutionUnits mMem mSteps
+
+_toTxExecUnits :: MaxTxExecutionUnits -> Alonzo.ExUnits
+_toTxExecUnits (MaxTxExecutionUnits (ExecutionUnits mMem mSteps)) =
+  Alonzo.ExUnits mMem mSteps
+
+instance ToJSON MaxTxExecutionUnits where
+  toJSON (MaxTxExecutionUnits (ExecutionUnits space time)) =
+    object [ "maxTxExecutionUnits" .=
+                object ["space" .= space, "time" .= time]
+           ]
+
+instance FromJSON MaxTxExecutionUnits where
+  parseJSON = withObject "MaxTxExecutionUnits" $ \o -> do
+    obj <- o .: "maxTxExecutionUnits"
+    MaxTxExecutionUnits
+      <$> (ExecutionUnits <$> obj .: "space" <*> obj .: "time")
+
+newtype MaxBlockExecutionUnits =
+    MaxBlockExecutionUnits { unMaxBlockExecutionUnits :: ExecutionUnits}
+    deriving (Eq, Show)
+
+_fromMaxBlockExec :: Alonzo.ExUnits -> MaxBlockExecutionUnits
+_fromMaxBlockExec (Alonzo.ExUnits mMem mSteps) =
+  MaxBlockExecutionUnits $ ExecutionUnits mMem mSteps
+
+_toExUnits :: MaxBlockExecutionUnits -> Alonzo.ExUnits
+_toExUnits (MaxBlockExecutionUnits (ExecutionUnits mMem mSteps)) =
+  Alonzo.ExUnits mMem mSteps
+
+instance ToJSON MaxBlockExecutionUnits where
+  toJSON (MaxBlockExecutionUnits (ExecutionUnits space time)) =
+    object [ "maxBlockExecutionUnits" .=
+                object ["space" .= space, "time" .= time]
+           ]
+
+instance FromJSON MaxBlockExecutionUnits where
+  parseJSON = withObject "MaxBlockExecutionUnits" $ \o -> do
+    obj <- o .: "maxBlockExecutionUnits"
+    MaxBlockExecutionUnits
+      <$> (ExecutionUnits <$> obj .: "space" <*> obj .: "time")
+
+data ExecutionUnits
+    = ExecutionUnits { space :: Word64
+                     , time :: Word64
+                     } deriving (Eq, Show)
+
+
+newtype CostModel = CostModel (Map AnyScriptLanguage Cost)
+                  deriving (Eq,Show)
+
+_fromLanguage :: Alonzo.Language -> AnyScriptLanguage
+_fromLanguage Alonzo.PlutusV1 = AnyScriptLanguage (PlutusScriptLanguage PlutusScriptV1)
+
+_fromCostModel :: Map Alonzo.Language Alonzo.CostModel -> CostModel
+_fromCostModel _ = CostModel mempty
+
+_toCostModel :: CostModel -> Map Alonzo.Language Alonzo.CostModel
+_toCostModel _ = Map.empty
+
+
+newtype Cost = Cost (Map.Map Operation Integer)
+             deriving (Eq, Show)
+
+data Operation = Add | Subtract | OtherOperation
+                 deriving (Eq, Ord, Show)
```

Comment:
So I don't think we'll provide an explicit representation for the cost model. We'll keep it opaque and wrap the types from the ledger, and just provide binary serialisation instances, not JSON I think.

Reply:
Updated: After speaking with Jared we concluded that we should expose a JSON format to users.